test/Microsoft.AspNet.Mvc.ModelBinding.Test/Validation/DefaultObjectValidatorTests.cs

```diff
@@ -472,19 +472,69 @@ public void NonRequestBoundModel_MarkedAsSkipped()
             validator.Validate(validationContext);
 
             // Assert
+            Assert.True(validationContext.ModelState.IsValid);
             var modelState = validationContext.ModelState["serviceProvider.TestService"];
             Assert.Empty(modelState.Errors);
             Assert.Equal(modelState.ValidationState, ModelValidationState.Skipped);
         }
 
+        [Fact]
+        public void EnumerableType_ValidatedSuccessfully()
+        {
+            // Arrange
+            var modelStateDictionary = new ModelStateDictionary();
+            modelStateDictionary.Add("items[0]", new ModelState());
+            modelStateDictionary.Add("items[1]", new ModelState());
+            modelStateDictionary.Add("items[2]", new ModelState());
+
+            var testValidationContext = GetModelValidationContext(
+                new string[] { "Foo", "Bar", "Baz" },
+                typeof(string[]),
+                "items",
+                excludedTypes: null,
+                modelStateDictionary: modelStateDictionary);
+
+            var excludedValidationTypesPredicate = new List<IExcludeTypeValidationFilter>();
+            excludedValidationTypesPredicate.Add(new SimpleTypesExcludeFilter());
+
+            var mockValidationExcludeFiltersProvider = new Mock<IValidationExcludeFiltersProvider>();
+            mockValidationExcludeFiltersProvider
+                .SetupGet(o => o.ExcludeFilters)
+                .Returns(excludedValidationTypesPredicate);
+            testValidationContext.ExcludeFiltersProvider = mockValidationExcludeFiltersProvider.Object;
+
+            var validationContext = testValidationContext.ModelValidationContext;
+
+            var validator = new DefaultObjectValidator(
+                testValidationContext.ExcludeFiltersProvider,
+                testValidationContext.ModelMetadataProvider);
+
+            // Act
+            validator.Validate(validationContext);
+
+            // Assert
+            Assert.True(validationContext.ModelState.IsValid);
+            var modelState = validationContext.ModelState["items"];
```

Comment:
Yes. This https://github.com/aspnet/Mvc/blob/9eb874b77c253818c6754ab779fdf50e9e9a1475/src/Microsoft.AspNet.Mvc.ModelBinding/ModelStateDictionary.cs#L395 will create a new `ModelState` for the key `items`.